python/src/aac/context/definition_parser.py

```diff
@@ -9,735 +9,828 @@
+            instance_class (Type): The class type of the instance being created.
 
-            Returns:
-                The populated instance for the given definition.
-            """
+        Returns:
+            The created instance list.
+        """
+        if is_list:
+            instance = []
+            if not field_value:
+                if is_required:
+                    raise LanguageError(f"Missing required field {field_name}")
+            else:
+                if not isinstance(field_value, list):
+                    if is_required:
+                        raise LanguageError(
+                            f"Invalid parsed value for field '{field_name}'.  Expected type 'list', but found '{type(field_value)}'.  Value = {field_value}",
+                            self.get_location_str(field_name, lexemes),
+                        )
+                    else:
+                        return instance
+                instance = self.field_instance_creator_list(lexemes, field_value, field_name, defining_definition, instance, instance_class)
+        else:
             instance = None
-
-            defining_definition = None
-            for item in context.get_definitions() + parsed_definitions:
-                if item.get_root_key() == "schema":
-                    if "root" in item.structure["schema"]:
-                        if (
-                            definition.get_root_key()
-                            == item.structure["schema"]["root"]
-                        ):
-                            defining_definition = item
-
-            if not defining_definition:
+            if not field_value:
+                if is_required:
+                    raise LanguageError(
+                        f"Missing required field {field_name}",
+                        self.get_location_str(field_name, lexemes),
+                    )
+                else:
+                    return instance
+            if not isinstance(field_value, dict):
+                # this is a complex type defined by a schema, with a field value so it should be a dict
                 raise LanguageError(
-                    f"Could not find definition for {definition.name} with root {definition.get_root_key()}",
-                    get_location_str(definition.get_root_key(), definition.lexemes),
+                    f"Invalid parsed value for field '{field_name}'.  Expected type 'dict', but found '{type(field_value)}'.  Value = {field_value}",
+                    self.get_location_str(field_name, lexemes),
                 )
 
+            # make sure there are no undefined fields
+            instance = self.field_instance_creator_not_list(lexemes, field_value, field_name, defining_definition, instance, instance_class)
+        return instance
+
+    def check_schema(self, is_list, field_value, field_name: str, is_required, lexemes, defining_definition):
```

Comment:
```suggestion
    def schema_field_value_check(self, is_list, field_value, field_name: str, is_required, lexemes, defining_definition) -> list:
```
Missing the return syntax and this is evaluating the field value so it should follow similar naming scheme to the primitive and enum field value check methods.